File: Programmers/두 개 뽑아서 더하기.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    for (int i = 0; i < numbers.size(); i++)
    {
        for (int j = i + 1; j < numbers.size(); j++)
        {
            answer.push_back(numbers[i] + numbers[j]);
        }
    }
    sort(answer.begin(), answer.end());
    answer.erase(unique(answer.begin(), answer.end()), answer.end());
    return answer;
}

int main() {
    vector<int> arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
    vector<int> test = solution(arr);

    for (int i = 0; i < test.size(); i++)
    {
        cout << test[i] << endl;
    }
    system("pause");
}

File: Programmers/두 개 뽑아서 더하기.java
import java.util.TreeSet;
class Solution {
    public int[] solution(int[] numbers) {
        TreeSet<Integer> ts = new TreeSet<>();
        for(int i = 0; i < numbers.length - 1; i++)
            for(int j = i + 1; j < numbers.length; j++)
                ts.add(numbers[i] + numbers[j]);
        int[] answer = new int[ts.size()];
         
        int tmp = 0;
        for(int i : ts){
            answer[tmp] = i;
            tmp++;
        }
        return answer;
    }
}